dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLIndexFilterUtils.java

```
@@ -92,29 +88,25 @@ public static String parseSortCriteria(@Nullable IndexSortCriterion indexSortCri
    */
   public static String parseIndexFilter(@Nonnull IndexFilter indexFilter) {
     List<String> sqlFilters = new ArrayList<>();
-    Set<String> aspectColumns = new HashSet<>(); // aspect columns (i.e. start with a_) to check if soft-deleted
+
     for (IndexCriterion indexCriterion : indexFilter.getCriteria()) {
-      final String aspect = indexCriterion.getAspect();
-      if (!(isUrn(aspect))) {
-        // urns, which are entities, cannot be soft-deleted so no need to check them
-        aspectColumns.add(getAspectColumnName(aspect));
+      final String aspect = indexCriterion.getAspect(GetMode.NULL);
```

no need for GetMode.NULL. `aspect` is a required field in IndexCriterion.pdl.

---

right, I think we should throw RequiredFieldNotPresentException if user did not set this field. Updated.